solar-system.rb

```diff
@@ -0,0 +1,197 @@
+#defines class solarsystem
+class SolarSystem
+
+
+ #helper method for @ planets, allows us to set and get @planets variable
+ attr_reader :planets
+ attr_accessor :planets_array
+ attr_accessor :user_planet
+
+#initialize method, will make it so calling an instance of SolarSystem class
+#requires the attribute initial_planets
+def initialize(initial_planets)
+    #instance variable @planets
+    @planets = initial_planets
+  end
+
+
+ def solarsystem_age
+   print "This Solar System is 4 days old."
+ end
+  # a summary method so that we can see what is going on with the
+  #instance variable
+  def summary
+    return "#{@planets}"
+  end
+
+
+  #method that prints out the instance variable as a list
+  def list_planets
+    n = 1
+    planets_array = []
+    #adds each instance of planet class to an array
+    @planets.each do |planet| #an instance of Planet class
+         planets_array << "#{n}.    #{planet.planet_name}"
+         #this is using the get method for planet to return the planet name
+         #as defined in the initialize
+         n += 1
+    end
+    #retunrs the planets array
+    return planets_array
+  end
+
+  #adds a planet to the instance variable of planets through user input
+  def add_planet
+     #allows user to input stats about the planet
+     print "What is the name of the new planet?"
+     planet_name = $stdin.gets.chomp.to_s
+     print "What is the diameter of the planet?"
+     planet_diameter = $stdin.gets.chomp.to_s
+     print "What is the mass of the planet relative to earth?"
+     mass_earth = $stdin.gets.chomp.to_i
+     print "How many moons does the planet have?"
+     number_of_moons = $stdin.gets.chomp.to_i
+     print "How many millions of kilometers is the planet from the sun?"
+     distance_from_the_sun = $stdin.gets.chomp.to_i
+     print "How long is it's year, relative to an earth year?"
+     earth_year = $stdin.gets.chomp.to_f
+
+
+     #creates a new instance of the planet and adds it to
+     #instance variable of Solar System class planets as a class of Planet
+     user_input_planet = Planet.new(planet_name,planet_diameter,mass_earth,number_of_moons, distance_from_the_sun, earth_year)
+       @planets.push(user_input_planet)
+
+     # the user can choose what step is next after adding a planet
+     user_choice
+  end
+
+#lists all planets that are available, and let's the user choose a planet
+#to read about
+  def user_choice
+     #for some reason list_planets would only work with a puts command
```

Comment:
I like that you've broken all this driver code out into separate methods. However, I'm not sure that `SolarSystem` should be responsible for interacting with the user like this. This design means that `SolarSystem` is responsible both for keeping track of a list of `Planet`s, and for running the control loop, which is a little confusing.

Instead, you might write a method on `SolarSystem` that takes a planet name and returns its info as a string, and another takes two names, calculates between them and returns it. Then you could write some separate driver code down below that calls those methods.